### DISCUSSION TITLE
How to get the ParseError from the auto-render extension?

### DISCUSSION THREAD

**Discussion by q2apro** [KaTeX/KaTeX#2616] [Q&A]
I am using the [Auto-render extension ](https://katex.org/docs/autorender.html) heavily ([github](https://github.com/KaTeX/KaTeX/tree/89daa02fd899d01b2a92f69ecd8be34bbece9d72/contrib/auto-render)).

I use the error handling like this:

	renderMathInElement(document.body, {
		delimiters: [
			{left: "$$", right: "$$", display: true},
			{left: "\\(", right: "\\)", display: false},
			{left: "\\[", right: "\\]", display: false},
		],
		macros: {
			":": "{\\char`:}",
			",": "{\\char`,}",
			"·": "{\\char`·}",
		},
		fleqn: true,
		errorCallback: function(error) 
		{
			// LATEX ERROR FOUND - Katex cannot parse
			// show error box bottom right
			var $div = $("<div>", {id: "dialog-box-error", "class": "error-bg" });
			// $div.click( function(){ $(this).remove() });
			$('.main').append($div);
			$div.text(error.replace('\\(', '').replace('\\)', '').replace(' with', ''));
			//console.log(error);
		},
		preProcess: (math) => math.replace(/&nbsp;/g, ' '),
	});
	console.log('Katex rendered');

With autorender I get (example):

> KaTeX auto-render: Failed to parse `a_N(150 \frac{ \mbox {km}}{\text{h}})=\frac{v^2}{r}=\frac{ \left(150 \frac{ \mbox {km}}{\text{h}} \right) ^2 }{200\text{m} }=  \frac{\left( 150\frac{1000\mbox {m}}{3600\text{s}} \right)^2 }{200\text{m}} \approx 8,681 \frac{\text{m}}{\text{s}^2}`KaTeX auto-render: Failed to parse `s=v_0 \cdot t + \frac{1}{2}a_T  \cdot t^2=120 \frac{\mbox {km}}{\text{h}} \cdot 3\text{s} + \frac{1}{2} 2,778\frac{\text{m}}{\text{s}^2} \cdot (3\text{s})^2=112,5\text{m}` with 

With Katex error handling, however, I get:

> KaTeX auto-render: Failed to parse `s=v_0 \cdot t + \frac{1}{2}a_T  \cdot t^2=120 \frac{\mbox {km}}{\text{h}} \cdot 3\text{s} + \frac{1}{2} 2,778\frac{\text{m}}{\text{s}^2} \cdot (3\text{s})^2=112,5\text{m}` with  ParseError: KaTeX parse error: Undefined control sequence: \mbox at position 53: … t^2=120 \frac{\̲m̲b̲o̲x̲ ̲{km}}{\text{h}}…

**The important part is missing:**

> ParseError: KaTeX parse error: Undefined control sequence: \mbox at position 53: … t^2=120 \frac{\̲m̲b̲o̲x̲ ̲{km}}{\text{h}}…

How to get the ParseError message in the callback?

**Comment by k4b7** [comment]
This will require code changes to the auto-render extension.  Can you file an issue for this?

**Reply by q2apro** [reply]
https://github.com/KaTeX/KaTeX/issues/2682

**Comment by jmooring** [comment]
The `errorCallback` below opens an alert window and displays the first line of the error stack, something like:

![image](https://github.com/user-attachments/assets/8ec80a28-000b-47a5-8a24-842eba06b80d)


```javascript
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.9/dist/katex.min.css" integrity="sha384-n8MVd4RsNIU0tAv4ct0nTaAbDJwPJzDEaqSD1odI+WdtXRGWt2kTvGFasHpSy3SV" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.9/dist/katex.min.js" integrity="sha384-XjKyOOlGwcjNTAIQHIpgOno0Hl1YQqzUOEleOLALmuqehneUG+vnGctmUb0ZY0l8" crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.9/dist/contrib/auto-render.min.js" integrity="sha384-+VBxd3r6XgURycqtZ117nYw44OOcIax56Z4dCRWbxyPt0Koah1uHoK0o4+/RRE05" crossorigin="anonymous"></script>
<script>
  document.addEventListener("DOMContentLoaded", () => {
    renderMathInElement(document.body, {
      delimiters: [
        { left: '\\[', right: '\\]', display: true }, // block
        { left: '$$', right: '$$', display: true },   // block
        { left: '\\(', right: '\\)', display: false }, // inline
      ],
      throwOnError: true,
      errorCallback: (msg, error) => alert(error.stack.split('\n')[0]),
    });
  });
</script>
```
